fairscale/experimental/nn/distributed_pipeline/partition_handler.py

```diff
@@ -275,14 +275,19 @@ def run_pipeline(self, pipeline_record_rref: rpc.RRef) -> Optional[Tensor]:
         pipeline_record = pipeline_record_rref.local_value()
         self.run(pipeline_record)
 
+        result: Optional[Tensor] = None
+
         if not pipeline_record.consumers:
-            result = microbatch.gather(pipeline_record.batches)
-            assert len(result) == 1
-            result = result[0]
+            gather_result = microbatch.gather(pipeline_record.batches)
```

Comment:
nit: `gather_results` sounds a bit better.

Reply:
hmmm, results might misleading in other ways. the function has only one object returning (though it is a list).

Reply:
Well, you assigned `gather_results[0]` to `result`, so it should be clear. It's up to you.